packages/arb-token-bridge-ui/src/components/TransactionHistory/TransactionHistory.tsx

```diff
@@ -124,39 +142,65 @@ export const TransactionHistory = ({
 
-  function handleSentOrReceivedTxForSCW(index: number) {
-    if (!isSmartContractWallet || !chain) {
-      return
-    }
-    const isDepositsTab = index === 0
-    const isConnectedToArbitrum = isNetwork(chain.id).isArbitrum
-    // SCW address is tied to a specific network, so we must ensure that:
-    if (isDepositsTab) {
-      // if showing deposits, we always show:
-      if (isConnectedToArbitrum) {
-        // - received txs if connected to L2
-        showReceivedTransactions()
-      } else {
-        // - sent txs if connected to L1
-        showSentTransactions()
+  const handleSmartContractWalletTxHistoryTab = useCallback(
+    (index: number) => {
+      if (!isSmartContractWallet || !chain) {
+        return
       }
-    } else {
-      // if showing withdrawals, we always show:
-      if (isConnectedToArbitrum) {
-        // - sent txs if connected to L2
-        showSentTransactions()
+      const isDepositsTab = index === TransactionHistoryTab.DEPOSITS
+      const isWithdrawalsTab = index === TransactionHistoryTab.WITHDRAWALS
+      const isConnectedToArbitrum = isNetwork(chain.id).isArbitrum
+      // SCW address is tied to a specific network, so we must ensure that:
+      if (isDepositsTab) {
+        // if showing deposits, we always show:
+        if (isConnectedToArbitrum) {
+          // - received txs if connected to L2
+          showReceivedTransactions()
+        } else {
+          // - sent txs if connected to L1
+          showSentTransactions()
+        }
+      } else if (isWithdrawalsTab) {
+        // Withdrawal tab
+        // if showing withdrawals, we always show:
+        if (isConnectedToArbitrum) {
+          // - sent txs if connected to L2
+          showSentTransactions()
+        } else {
+          // - received txs if connected to L1
+          showReceivedTransactions()
+        }
       } else {
-        // - received txs if connected to L1
-        showReceivedTransactions()
+        // Cctp tab
+        if (isConnectedToArbitrum) {
+          showCctpDepositsTransactions()
+        } else {
+          showCctpWithdrawalsTransactions()
+        }
       }
-    }
-  }
+    },
+    [
+      chain,
+      isSmartContractWallet,
+      showCctpDepositsTransactions,
+      showCctpWithdrawalsTransactions,
+      showReceivedTransactions,
+      showSentTransactions
+    ]
+  )
 
   useEffect(() => {
     // this function runs every time the network tab is changed, and here it is also triggered when the page loads
     // it sets the tab to 0 (deposits), which is the default tab
-    handleSentOrReceivedTxForSCW(0)
-  }, [isSmartContractWallet, chain])
+    handleSmartContractWalletTxHistoryTab(0)
+  }, [handleSmartContractWalletTxHistoryTab])
+
+  useEffect(() => {
+    // This check avoid the situation when we open the transaction history on CCTP tab after a transfer, but it would go back to "To Arbitrum" tab
+    if (transfersIds.length === 0) {
```

Comment:
Not sure if it's to do with this line, but now when you try to e.g. deposit ETH, it doesn't change the tab to 'To Arbitrum' but keeps the currently active tab open.

Reply:
Updated, I have added logic to open the tx history on the proper tab in `TransferPanel`